dds/DCPS/security/CryptoBuiltInImpl.h

```diff
@@ -77,6 +77,11 @@ class DdsSecurity_Export CryptoBuiltInImpl
     bool relay_only,
     DDS::Security::SecurityException& ex);
 
+  virtual bool rekey_remote_datareader(
```

Comment:
Could this method be removed and "folded in" to register_matched_remote_datareader?

Reply:
To do so, we would need to commit to either 1) caching the DatareaderCryptoHandle in a way that we can find it again on the next call to register_matched_remote_datareader or 2) generate a new crypto handle and distribute it everywhere that it needs to be.

Reply:
Just to confirm, this method is only used for the BuiltinParticipantVolatileMessageSecure endpoints, correct?

I was not thinking of case (2) and it doesn't seem like what we want.

I think (1) isn't difficult and we should do it.  I could work on this if you'd like.